### ISSUE TITLE
while in F5 savegame dialog, CPU usage goes up to 100%

### ISSUE THREAD

**Issue by rofl0r** [adventuregamestudio/ags#750]
noticed in 1213 SE

i guess the loop displaying the save thingy needs a msleep(20) / SDL_Delay(20); equivalent, and optimally it wouldnt only repaint if something changed.

**Comment by sonneveld** [comment]
To whoever gets to this issue, please don't use delays to fix this.

**Comment by sonneveld** [comment]
Could you please check if https://github.com/adventuregamestudio/ags/pull/777 fixes this issue as well?

**Comment by sonneveld** [comment]
Some of the dialogues do have their own weird little loops, is this a specific game you're seeing this in?

This should be fixed in `ags3` now if you'd like to test it.  Still waiting on PR to update the 3.5.0 release branch.

**Comment by sonneveld** [comment]
Closing because duplicate of https://github.com/adventuregamestudio/ags/issues/715

**Comment by rofl0r** [comment]
> Closing because duplicate of #715

this is unrelated, it happened with 3.4.x

> is this a specific game you're seeing this in

yes, `1213 SE` as mentioned initially

> This should be fixed in ags3 now if you'd like to test it

nope, still not fixed. please reopen.

**Comment by ivan-mogilko** [comment]
> this is unrelated, it happened with 3.4.x

TBH I also did not know this happens in 3.4.x, thought this happens in recent ags3 branch.

**Comment by tzachshabtay** [comment]
You can create [an issue template](https://help.github.com/en/articles/creating-issue-templates-for-your-repository) with a simple checklist to tell people who open issues about what's important to you to know, i.e something like:
```
Please tell us (if applicable) about the environment in which you saw the bug:

AGS Version:

Operating System:

```

**Comment by rofl0r** [comment]
> You can create an issue template with a simple checklist 

uh, i assumed that was clear since i opened 2 or 3 other issues that same day or week, each time stating that i used 3.4.x release. no need to introduce bureaucracy.

**Comment by ivan-mogilko** [comment]
> uh, i assumed that was clear since i opened 2 or 3 other issues that same day or week, each time stating that i used 3.4.x release

Maybe, I could simply forget. The problem is these things are clear so long as you remember them, but not if you return back to existing issue after a while, then you may no longer remember what it relates to.
Imho at least the version tag/branch would be useful to have.

**Comment by sonneveld** [comment]
> uh, i assumed that was clear since i opened 2 or 3 other issues that same day or week, each time stating that i used 3.4.x release. no need to introduce bureaucracy.

It was a month ago. Looking at this ticket in isolation I assumed this was a general ags3 issue.

> nope, still not fixed. please reopen.

Do you mean still not fixed in 3.5 as well?

**Comment by sonneveld** [comment]
I think I can confirm it's not an issue in `ags3`

![image](https://user-images.githubusercontent.com/335451/58741465-31ddfc00-845c-11e9-95f8-882352452efd.png)

Using 1213 SE episode 1 as a test, from here: http://www.g4g.it/download/comment.php?dlid=3158

**Comment by rofl0r** [comment]
> Do you mean still not fixed in 3.5 as well?

correct. the game i just tested, terror of the vampire (https://www.adventuregamestudio.co.uk/site/games/game/2170), exhibited the issue too.

**Comment by sonneveld** [comment]
What changeset of `ags3` are you using?

**Comment by rofl0r** [comment]
> I think I can confirm it's not an issue in ags3

does windows task manager divide up cpu load over the number of cores ?

**Comment by sonneveld** [comment]
> > I think I can confirm it's not an issue in ags3
> 
> does windows task manager divide up cpu load over the number of cores ?

Correct, it's a little annoying. So it's 2.6/12.5

**Comment by sonneveld** [comment]
If you'd like to create a PR for 3.4, the offending line is here: https://github.com/adventuregamestudio/ags/blob/release-3.4.0/Engine/gui/cscidialog.cpp#L193

`        while (timerloop == 0) ;`

**Comment by rofl0r** [comment]
hmm, seems to be partially fixed indeed in latest `ags3`. 1213 SE cpu load is now way down, whereas "terror of the vampires" still uses 70% while the save dialog is opened.

> edit: note this will still be a problem in those "revert new timing" pull requests.

well, that PR is moot anyway since you already pushed your changes...

**Comment by sonneveld** [comment]
> hmm, seems to be partially fixed indeed in latest `ags3`. 1213 SE cpu load is now way down, whereas "terror of the vampires" still uses 70% while the save dialog is opened.

Terror of the vampire uses software rendering for the palette plugin. I had to enable optimisations to get a decent cpu usage (it was at 16% or something silly before):
![image](https://user-images.githubusercontent.com/335451/58741621-7ff3ff00-845e-11e9-946f-1a293118e571.png)

**Comment by sonneveld** [comment]
> well, that PR is moot anyway since you already pushed your changes...

That's not true. The `release-3.5.0` branch still needs _something_ done to fix the timing issues.

**Comment by sonneveld** [comment]
> You can create [an issue template](https://help.github.com/en/articles/creating-issue-templates-for-your-repository) with a simple checklist to tell people who open issues about what's important to you to know, i.e something like:

Not to add to bureaucracy but I kind of like this idea.  changeset if build from repository, game, link to game download if possible, save game to the point of issue, etc.  Just general reminders for questions we probably will ask anyway.

**Comment by ivan-mogilko** [comment]
So, this is built-in dialogs that have their own update/waiting loop.

**Comment by ivan-mogilko** [comment]
I pushed a commit: 8f43984b8d65ccbb4f1bae1c312fc4cd7154d137

**Comment by rofl0r** [comment]
thanks, though to which branch ?

**Comment by ivan-mogilko** [comment]
Oh, sorry, it's in "release-3.4.4" branch.

**Comment by rofl0r** [comment]
odd, i can't see it in the commit list: https://github.com/adventuregamestudio/ags/commits/release-3.4.4

**Comment by ivan-mogilko** [comment]
I did a mistake and pushed it to my personal fork first, now it should be here too.

**Comment by ivan-mogilko** [comment]
I tested and it seem to have fixed the cpu load. Now there is a patch released (github only, I was not planning to release on forums because there will be 3.5.0 release soon anyway): https://github.com/adventuregamestudio/ags/releases/tag/v.3.4.4.1

**Comment by rofl0r** [comment]
thanks, i didn't get around yet to recompile 3.4.x branch as i have a ton of custom patches on top of 3.5.x, but i hope i'll have time to compile/test the 3.4.4.1 release soon
